src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionAttachmentManager.cs

```diff
@@ -290,7 +290,12 @@ private void AddNewFileTransfer(FileTransferInformation fileTransferInfo, AsyncC
                         ex.ToString(),
                         uri,
                         friendlyName,
-                        Guid.Parse(testCaseId));
+#if NET7_0_OR_GREATER
+                        Guid.Parse(testCaseId, CultureInfo.InvariantCulture)
+#else
+                        Guid.Parse(testCaseId)
+#endif
```

Comment:
Not sure if that's way better but an alternative is

```suggestion
                        Guid.Parse(testCaseId
#if NET7_0_OR_GREATER
                            , CultureInfo.InvariantCulture
#endif
                            )
```